src/main/java/jcifs/http/NtlmHttpFilter.java

```diff
@@ -0,0 +1,380 @@
+ * complete details.
+ * 
+ * @deprecated NTLMv1 only
+ */
+@Deprecated
+public class NtlmHttpFilter implements Filter {
+
+    private static final Logger log = LoggerFactory.getLogger(NtlmHttpFilter.class);
+
+    private String defaultDomain;
+    private String domainController;
+    private boolean loadBalance;
+    private boolean enableBasic;
+    private boolean insecureBasic;
+    private String realm;
+
+    private CIFSContext transportContext;
+    private Address[] dcList = null;
+    private long dcListExpiration;
+
+    private int netbiosLookupRespLimit = 3;
+    private long netbiosCacheTimeout = 60 * 60 * 10;
+    private static int dcListCounter;
+
+
+    @Override
+    public void init ( FilterConfig filterConfig ) throws ServletException {
+        String name;
+
+        Properties p = new Properties();
+        /*
+         * Set jcifs properties we know we want; soTimeout and cachePolicy to 30min.
+         */
+        p.setProperty("jcifs.smb.client.soTimeout", "1800000");
+        p.setProperty("jcifs.netbios.cachePolicy", "1200");
+        /*
+         * The Filter can only work with NTLMv1 as it uses a man-in-the-middle
+         * technique that NTLMv2 specifically thwarts. A real NTLM Filter would
+         * need to do a NETLOGON RPC that JCIFS will likely never implement
+         * because it requires a lot of extra crypto not used by CIFS.
+         */
+        p.setProperty("jcifs.smb.lmCompatibility", "0");
+        p.setProperty("jcifs.smb.client.useExtendedSecurity", "false");
+
+        Enumeration<String> e = filterConfig.getInitParameterNames();
+        while ( e.hasMoreElements() ) {
+            name = e.nextElement();
+            if ( name.startsWith("jcifs.") ) {
+                p.setProperty(name, filterConfig.getInitParameter(name));
+            }
+        }
+
+        try {
+            this.defaultDomain = p.getProperty("jcifs.smb.client.domain");
+            this.domainController = p.getProperty("jcifs.http.domainController");
+            if ( this.domainController == null ) {
+                this.domainController = this.defaultDomain;
+                this.loadBalance = Config.getBoolean(p, "jcifs.http.loadBalance", true);
+            }
+            this.enableBasic = Boolean.valueOf(p.getProperty("jcifs.http.enableBasic")).booleanValue();
+            this.insecureBasic = Boolean.valueOf(p.getProperty("jcifs.http.insecureBasic")).booleanValue();
+            this.realm = p.getProperty("jcifs.http.basicRealm");
+            this.netbiosLookupRespLimit = Config.getInt(p, "jcifs.netbios.lookupRespLimit", 3);
+            this.netbiosCacheTimeout = Config.getInt(p, "jcifs.netbios.cachePolicy", 60 * 10) * 60; /* 10 hours */
+
+            if ( this.realm == null )
+                this.realm = "jCIFS";
+
+            this.transportContext = new BaseContext(new PropertyConfiguration(p));
+        }
+        catch ( CIFSException ex ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)